File: databasyrepo/databasyrepo/models/core/tests/basic_tests.py
import json
from unittest import TestCase
from databasyrepo.models.core.models import Model
from databasyrepo.models.core.reprs import Canvas
from databasyrepo.models.core.serializing import deserialize
from databasyrepo.models.register import register

__author__ = 'Marboni'

class BasicTest(TestCase):
    def test_register(self):
        model_class = register.by_key(Model.code())
        self.assertEqual(Model, model_class)

        with self.assertRaises(ValueError):
            register.by_key('non.existent.class')
        with self.assertRaises(ValueError):
            register.by_key(Model.code(), Canvas)

    def test_serialization(self):
        model = Model.create(1L, 1L)
        serialized_model = json.dumps(model)
        deserialized_model = deserialize(serialized_model)

        self.assertEqual(model, deserialized_model)

File: databasyrepo/databasyrepo/static/repo/src/databasy/model/core/tests/basic_tests.js
test('basic_tests : test_register', function() {
    var register = databasy.model.register.register;
    var model_class = register.by_key(databasy.model.core.models.Model.CODE);
    equal(databasy.model.core.models.Model, model_class);

    throws(function() {
        register.by_key('non.existent.code')
    });
});

test('basic_tests : test_serialization', function() {
    var model = databasy.model.core.models.Model.createModel(1, 10);
    var serialized_model = model.serialize();
    var deserialized_model = databasy.model.core.serializing.Serializable.deserialize(serialized_model);

    deepEqual(deserialized_model, model);
});